flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -296,6 +299,39 @@ class CodeGenerator(
       fields.mkString(", ")
     }
 
+    def genInitialize(): String = {
+      
+      val sig: String = 
+        j"""
+           |  public void initialize(
+           |    org.apache.flink.api.common.functions.RuntimeContext ctx
+           |  )""".stripMargin
+        
+      val initDist: String = if( distinctAggsFlags.isDefined ) {
+        val statePackage = "org.apache.flink.api.common.state"
+        val distAggsFlags = distinctAggsFlags.get
+          for(i <- distAggsFlags.indices) yield
+            if(distAggsFlags(i)) {
+              val typeString = javaTypes(aggFields(i)(0))
```

Comment:
That's actually a very good point! We have to use `Row` because `Tuple` doesn't support `null` values.